common/metrics/config.go

```diff
@@ -399,3 +443,64 @@ func configExcludeTags(cfg ClientConfig) map[string]map[string]struct{} {
 	}
 	return tagsToFilter
 }
+
+func (s SanitizeRange) toTally() (tally.SanitizeRange, error) {
+	startRangeRunes := []rune(s.StartRange)
+	if len(startRangeRunes) != 1 {
+		return tally.SanitizeRange{}, fmt.Errorf("start range '%+v' must be a single rune", startRangeRunes)
+	}
+
+	endRangeRunes := []rune(s.EndRange)
+	if len(endRangeRunes) != 1 {
+		return tally.SanitizeRange{}, fmt.Errorf("end range '%+v' must be a single rune", endRangeRunes)
+	}
+
+	return tally.SanitizeRange([2]rune{startRangeRunes[0], endRangeRunes[0]}), nil
+}
+
+func (v ValidCharacters) toTally() (tally.ValidCharacters, error) {
+	var ranges []tally.SanitizeRange
+
+	for _, r := range v.Ranges {
+		tallyRange, err := r.toTally()
+		if err != nil {
+			return tally.ValidCharacters{}, err
+		}
+
+		ranges = append(ranges, tallyRange)
+	}
+
+	return tally.ValidCharacters{
+		Ranges:     ranges,
+		Characters: []rune(v.SafeCharacters),
+	}, nil
+}
+
+func (s SanitizeOptions) toTally() (tally.SanitizeOptions, error) {
+	tallyNameChars, err := s.NameCharacters.toTally()
+	if err != nil {
+		return tally.SanitizeOptions{}, errors.New("invalid nameChars")
+	}
+
+	tallyKeyChars, err := s.KeyCharacters.toTally()
+	if err != nil {
+		return tally.SanitizeOptions{}, errors.New("invalid keyChars")
+	}
+
+	tallyValueChars, err := s.ValueCharacters.toTally()
+	if err != nil {
+		return tally.SanitizeOptions{}, errors.New("invalid valueChars")
+	}
```

Comment:
the error should wrap the detailed error.

Reply:
oops, good catch